apps/internal/oauth/resolvers.go

```diff
@@ -119,9 +119,15 @@ func (m *authorityEndpoint) addCachedEndpoints(authorityInfo authority.Info, use
 	m.cache[authorityInfo.CanonicalAuthorityURI] = updatedCacheEntry
 }
 
-func (m *authorityEndpoint) openIDConfigurationEndpoint(ctx context.Context, authorityInfo authority.Info, userPrincipalName string) (string, error) {
-	if authorityInfo.Tenant == "adfs" {
+func (m *authorityEndpoint) openIDConfigurationEndpoint(ctx context.Context, authorityInfo authority.Info) (string, error) {
+	if authorityInfo.AuthorityType == authority.ADFS {
 		return fmt.Sprintf("https://%s/adfs/.well-known/openid-configuration", authorityInfo.Host), nil
+	} else if authorityInfo.AuthorityType == authority.DSTS {
+		resp, err := m.rest.Authority().DSTSInstanceDiscovery(ctx, authorityInfo)
```

Comment:
>DSTSInstanceDiscovery [](http://example.com/codeflow?start=34&length=21)

Similar to ADFS, you can use a hard coded URL for dSTS. An example of this endpoint on dSTS

[https://test-instance1-dsts.dsts.core.azure-test.net/dstsv2/7a433bfc-2514-4697-b467-e0933190487f/.well-known/openid-configuration](https://test-instance1-dsts.dsts.core.azure-test.net/dstsv2/7a433bfc-2514-4697-b467-e0933190487f/.well-known/openid-configuration)

IMPORTANT! To connect to this endpoint you need to be on our VPN.

Reply:
hard-coded the url